plugin/supervisor.go

```diff
@@ -17,9 +19,11 @@ import (
 )
 
 type supervisor struct {
-	client      *plugin.Client
-	hooks       Hooks
-	implemented [TotalHooksId]bool
+	client         *plugin.Client
+	clientProtocol plugin.ClientProtocol
```

Comment:
Why store this and not call `client.Client()`?

Reply:
> Why store this and not call client.Client()?

Good call. I had missed the portion of the `Client()` [method](https://github.com/hashicorp/go-plugin/blob/3f118e8ee104b6f22aeb12453fab56aed1356186/client.go#L349) that short circuits if the `ClientProtocol` is already assigned.